lint-rules.md

```diff
@@ -0,0 +1,1281 @@
+
+If there is a singular input and it can fit on that same line while being equal to or under the 90 character width limit, all tokens should be separated by a single space, likeso: `call foo_task { input: bam = sorted_bam }`. Note that there is no trailing comman here. This could optionally be split into 3 lines, likeso:
+
+```wdl
+call foo_task { input:
+    bam = sorted_bam,
+}
+```
+
+Note the trailing comma.
+
+**Group**: `spacing`
+
+#### Example
+
+Good (note that these examples take advantage of the input simplifying mechanism made available in the [v1.1 specification](https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#call-statement)):
+
+```wdl
+call md5sum.compute_checksum { input: file = bam }
+
+call samtools.quickcheck { input: bam }
+
+call util.compression_integrity { input:
+    bgzipped_file = bam,
+}
+
+call markdups_post_wf.markdups_post { input:
+    markdups_bam = select_first([
+        markdups.duplicate_marked_bam,
+        "undefined",
+    ]),
+    markdups_bam_index = select_first([
+        markdups.duplicate_marked_bam_index,
+        "undefined",
+    ]),
+    coverage_beds,
+    coverage_labels = parse_input.labels,
+    prefix = post_subsample_prefix + ".MarkDuplicates",
+}
+```
+
+### `snake_case`
+
+Variables, tasks, and workflows should be in lowercase "snake_case".
+
+**Group**: `naming`
+
+### `pascal_case`
+
+Declared structs should be in "PascalCase".
+
+**Group**: `naming`
+
+### `indentation`
+
+Indentation should be 4 spaces and never tab literals.
+
+### `trailing_whitespace`
+
+No whitespace on empty lines. No whitespace at the end of lines.
+
+### `one_empty_line`
+
+At most one empty line in a row. There should never be 2 or more blank lines in a row.
+
+**Group**: `spacing`
+
+### `no_newline_eof`
+
+The file must end with a newline.
+
+**Group**: `spacing`
+
+### `comment_whitespace`
+
+Comments on the same line as code should have 2 spaces before the `#` and one space before the comment. Comments on their own line should match the indentation level around them and have one space between the `#` and the comment. Keep in mind that even comments must be kept below the 90 character width limit.
+
+**Group**: `spacing`
+
+#### Example
+
+Good:
+
+```wdl
+    input {
+        Int x  # TODO better names here?
+        Int y
+    }
+
+    # Next do some math
+    Int z = x + y  # simple addition
+```
+
+Bad:
+
+```wdl
+    input {
+        Int x # Only one space before the comment. There should be two.
+        Int y  #There should be a space following the pound sign before the text.
+    }
+
+# This comment has an incorrect level of indentation
+    Int z = x + y# NO spaces between code and comment is hard to read.
+```
+
+### `double_quotes`
+
+All quotes should be double quotes.
+
+**Group**: `naming`
```

Comment:
This got lumped under "naming" because I went about the group concept by asking "what would likely be disabled at the same time?" I think double quotes and snake_case etc. rules make sense together from that perspective.